client/internal/statemachine/retry.go

```diff
@@ -27,3 +29,23 @@ func UntilStarted(ctx context.Context, client http.Client, deviceUid string, sta
 		return false, nil
 	}
 }
+
+// UntilDoneByIdentifier polls a state machine by its identifier until it has state done, or ends early due to error.
+// Note if you know a state machine with the same identifier has been run before, it may be checking the old state machine.
+func UntilDoneByIdentifier(ctx context.Context, client http.Client, stateMachineName string) retry.Func {
+	return func() (bool, error) {
+		sm, err := ReadInstanceByName(ctx, client, NewReadInstanceByNameInput(stateMachineName))
+		if err != nil {
+			return false, err
+		}
+		client.Logger.Println(fmt.Sprintf("state machine state=%s", sm.StateMachineInstanceCondition))
+		if sm.StateMachineInstanceCondition == state.ERROR {
+			if sm.StateMachineDetails.LastError != nil {
+				return false, fmt.Errorf(fmt.Sprintf("state machine errored: %s", sm.StateMachineDetails.LastError.ErrorMessage))
+			} else {
+				return false, fmt.Errorf("state machine errored (no error message)")
+			}
+		}
+		return sm.StateMachineInstanceCondition == state.DONE, nil
+	}
+}
```

Comment:
it is not used at the moment, but was used for experimenting, keeping because it is very likely to be used again in the future

Reply:
Please let's not bring it in this PR; it will just confuse us in the future.

Reply:
Sure removed 👍